### ISSUE TITLE
Heartbleed not detected

### ISSUE THREAD

**Issue by ericrange** [nabla-c0d3/sslyze#202]
Seems that the heartbleed plugin does not detect a heartbleed vuln. correctly.

example site: ascn.papasg.com

![image](https://cloud.githubusercontent.com/assets/1785416/24493639/c9cd04c8-152f-11e7-9ef8-8d4b51da10fe.png)

**Comment by nabla-c0d3** [comment]
Thanks - this was a timing issue.

**Comment by al3xdelarge** [comment]
This is unfortunately again an issue.
Heartbleed and ccs injection are not detected correctly in version 2.0.6.

I can privatly share a vulnerable system with you if needed.

**Comment by nabla-c0d3** [comment]
@al3xdelarge can you send it to my email address (on my GH profile)? Thanks!

**Comment by al3xdelarge** [comment]
Just sent it. Thank you!

**Comment by nabla-c0d3** [comment]
@al3xdelarge I finally looked into it. This was due to the server's TLS stack not supporting the Server Name Indication extension (or just rejecting any name sent via SNI). 

Since SSLyze always sends the SNI extension, the server was always returning a TLS Alert 112 "unrecognized_name" (ie. rejecting the SNI extension's value), ending the TLS handshake before SSLyze could test for Heartbleed and CCS.

The fix I just committed was to remove the SNI extension from all checks (including Heartbleed and CCS) except the certificate_info check which requires it. With this changes, SSLyze correctly flags the server you sent me as vulnerable.
